lib/eth/abi/event.rb

```diff
@@ -24,20 +24,69 @@ module Abi
     module Event
       extend self
 
+      # Compute topic for ABI event interface.
+      #
+      # @param interface [Hash] ABI event interface.
+      # @return [String] a hex-string topic.
+      def compute_topic(interface)
+        sig = Abi.signature(interface)
+        Util.prefix_hex(Util.bin_to_hex(Util.keccak256(sig)))
```

Comment:
Wondering if this `Util.prefix_hex(Util.bin_to_hex(Util.keccak256(sig)))` should be it's own helper? ethers.js calls it `ethers.utils.id("foo(address)")`

Reply:
This entire back-and-forth between binary and hexadecimal is a pain point, I have to agree.

I recently wrote a crystal library where I decided to create my own numeric data type `Num` that is a binary, decimal, and hexadecimal representation at the same time. So in that case you could just return `Util.keccak256(sig).to_prefixed_hex` or `.to_zpadded_hex` and so on.

https://github.com/q9f/secp256k1.cr/blob/58e955ce160bb5d15c0de3eecbf73bf06520be08/src/secp256k1/num.cr#L129

Maybe, it's worth having something like this for the Ruby library too.

Reply:
> Maybe, it's worth having something like this for the Ruby library too.

👍🏻 Yeah, definitely.

Reply:
Would porting the `Num` class over from the above link be a good first issue for me to work on? Is it already being worked on?

Reply:
Feel free to grab the Num class :)